### PR TITLE
Feature/meta data

### PR DESCRIPTION
**Pull request by rnugent3**

### CODE REVIEW

File: fda-model/paireddata/UncertainPairedData.cs [hunk 1]
```diff
@@ -139,7 +139,7 @@ public IPairedData SamplePairedData(double probability)
                 }
                 if (pairedData.RuleMap[nameof(pairedData.Xvals)].ErrorLevel > ErrorLevel.Unassigned)
                 {
-                    Array.Sort(pairedData.Xvals);//bad news.
+                    throw new Exception("X values are not monotonically increasing");
```
Comment (HenryGeorgist) [inline fda-model/paireddata/UncertainPairedData.cs:142]: im not a fan of exceptions. can we use the message system?

another possiblity is "force monotonic in range" to make it increase.

File: fda-model/paireddata/GraphicalUncertainPairedData.cs [hunk 2]
```diff
@@ -151,7 +159,7 @@ public IPairedData SamplePairedData(double probability)
                 }
                 if (pairedData.RuleMap[nameof(pairedData.Xvals)].ErrorLevel > ErrorLevel.Unassigned)
                 {
-                    Array.Sort(pairedData.Xvals);//bad news.
+                    throw new Exception("");
```
Comment (HenryGeorgist) [inline fda-model/paireddata/GraphicalUncertainPairedData.cs]: if you are going to do an exception we need to give it context - so add a message at least

**Review by HenryGeorgist — CHANGES_REQUESTED** [review]
please minimally fix the exception with an empty string.

File: fda-model/paireddata/UncertainPairedData.cs (hunk 1 above)
Comment (rnugent3) [reply fda-model/paireddata/UncertainPairedData.cs:142]: I will give it a try

**Comment by rnugent3** [comment]
the exception with an empty string was from an old commit and was previously replaced by an exception with a message, which has now been replaced with an IReportMessage Message

**Review by HenryGeorgist — COMMENTED** [review]
changes look good - now the only problem is that we dont fix the issue. So the compute keeps going when we identify this issue.

**Comment by rnugent3** [comment]
My thinking was to throw the exception and stop the compute because the data will have gone through two rounds of validation and then got messed up. should we just force monotonic and relay the same message as with y vals?

**Comment by rnugent3** [comment]
so then we'd need force monotonic in Y and force monotonic in X

**Comment by rnugent3** [comment]
I am going to merge this to continue work on a fresh branch and can resolve the non-monotonic X values on another branch

**Pull request merged**